lib/streamlit/elements/vega_charts.py

```diff
@@ -128,13 +131,21 @@ def _marshall_chart_data(
     These operations will happen in-place."""
 
     # Pull data out of spec dict when it's in a 'datasets' key:
-    #   datasets: {foo: df1, bar: df2}, ...}
+    #   datasets: {foo: df1_bytes, bar: df2_bytes}, ...}
     if "datasets" in spec:
         for dataset_name, dataset_data in spec["datasets"].items():
             dataset = proto.datasets.add()
             dataset.name = str(dataset_name)
             dataset.has_name = True
-            dataset.data.data = _serialize_data(dataset_data)
+            # The ID transformer already serializes the data into Arrow IPC format (bytes)
+            # If its already in bytes, we don't need to serialize it again.
```

Comment:
Great to see a comment here! I thin it could be made even better by adding some more context to it. What is the `ID transformer` (the `id_transform` function?) and how is it related to `_marshall_chart_data`? Are they supposed to call in a sequence?